streaming/java/streaming-api/src/main/java/io/ray/streaming/operator/chain/ChainedOperator.java

```diff
@@ -83,6 +84,23 @@ public Operator getTailOperator() {
     return tailOperator;
   }
 
+  @Override
+  public Object doCheckpoint(long checkpointId) {
+    Object[] checkpoints = new Object[operators.size()];
+    for (int i = 0; i < operators.size(); ++i) {
+      checkpoints[i] = operators.get(i).doCheckpoint(checkpointId);
+    }
+    return checkpoints;
+  }
+
+  @Override
+  public void loadCheckpoint(Object checkpointObject, long checkpointId) {
+    Object[] checkpoints = (Object[]) checkpointObject;
```

Comment:
It's so strange to convert object to a object array.

Reply:
I think it's ok, user won't be aware of this.